src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -2552,12 +2543,15 @@ bool LibraryCallKit::inline_vector_convert() {
       op = gvn().transform(VectorCastNode::make(cast_vopc, op, elem_bt_to, num_elem_to));
     } else { // num_elem_from == num_elem_to
       if (is_mask) {
-        if ((dst_type->isa_vectmask() && src_type->isa_vectmask()) ||
-            (type2aelembytes(elem_bt_from) == type2aelembytes(elem_bt_to))) {
-          op = gvn().transform(new VectorMaskCastNode(op, dst_type));
-        } else {
-          op = VectorMaskCastNode::makeCastNode(&gvn(), op, dst_type);
+        // Make sure that cast for vector mask is implemented to particular type/size combination.
+        if (!arch_supports_vector(Op_VectorMaskCast, num_elem_to, elem_bt_to, VecMaskNotUsed)) {
```

Comment:
`VecMaskUseAll` is fine to me since the final vector box may need to store mask if it is mask casting/reinterpreting. So since it has checked the load/store mask on Line#2439, do you think it needs to double check the load mask pattern here?

Reply:
Since checks related to VecMaskUseAll are decoupled from the main opcode so one check per intrinsic routine should be ok if its covering the needed trajectories. Infact we are passing VecMaskUseAll back to back to two rountines of Line #2439 and Line # 2452, WDYT ?

Reply:
Yes, I'm agree that one check is enough.

Reply:
I'm sorry that the second check on Line-2452 should be kept since the element type and vector length might be different from the first check on Line-2439. So I'd like to keep it as it is.